### ISSUE TITLE
IP Limit changed? Softbans?

### ISSUE THREAD

**Issue by Coalado** [seikur0/PGO-mapscan-opt#283]
I used to plan scan and divide each cell in 2. Then scan with 100-150 accounts.
This setup used to work fine.
Since a few days, it does not work any more.  After a few minutes all cells return as empty. 

The Planscanner continues anyway and writes the learning file. It does not complain at all. 
The learning file contains stops and gyms, but hardly any spawn points.

Would it be possible to detect this ind of ban?

**Comment by c00ni** [comment]
Same happened to me, but just during iscan.
Was running 80 accounts and worked fine for about 12 hours.
Suddenly started getting non-empty cells returned as empty. Was IP softbanned. I can indeed see how this is a much bigger problem for plan scanning.

**Comment by seikur0** [comment]
What's in the quality section of the spawnpoint file? It should be pretty clear from that, that a softban occurred. Other than that it'd be helpful if you can find out if it's due to having too many requests at once (too many accounts), having the accounts too spread out (too much spread), the scan delay of 10 seconds being to low in general now (time_hb in main0.py) or other reasons.

**Comment by Coalado** [comment]
Cell/4 and 50 accounts works fine.. will do further testing

**Comment by chelming** [comment]
@c00ni how large of an area? Level 10 S2 cell?

@Coalado I'm seeing the same as you. Cell/4 with 50 accounts and no issues.

**Comment by darkforcesjedi** [comment]
When I set up my scans I ensure the maximum dimension of my scan area is always less than 20km and I haven't run into soft bans since. I originally tried with my whole map in one profile (max dimension about 50km) and occasionally got soft bans after running the map for a long period of time.

From experimenting it seems that a single long distance teleport doesn't immediately trigger a ban, but a second one sometimes (but not always) does.

**Comment by chelming** [comment]
@darkforcesjedi which dimension? Largest point-to-point distance? Radius? How many accounts?

**Comment by darkforcesjedi** [comment]
Max distance between scan locations. Number of accounts I usually use {scan locations}/250.

**Comment by darkforcesjedi** [comment]
I am currently running a map with an island over 20km long and less than 1km wide using iscan and 7 accounts without issue (there are only like 1400 spawn points on the island).

**Comment by chelming** [comment]
@c00ni I spoke too soon. When I got to my 4th or 5th S2 cell, I got a soft IP ban using 50 accounts and having the S2 cell split in 4.

**Comment by darkforcesjedi** [comment]
Are you using a shared connection (open proxy, VPN, etc)? There could be other people using that connection with bots or scanners. I am using 140 accounts spread across 3 IPs at the moment and I haven't hit any softbans in weeks.

**Comment by c00ni** [comment]
@cwhits it was an area of approximately 2 whole level 10 cells. It was a bit more than 20k spawn points. The 80 accounts failed to catchup so I had to crop a few thousand out.

My ban reasons can be varied though:

It might have to do with me encountering some Pokémon types and I've even modified it to not use a 10 second delay for it (though this works fine all the time except this instance, including for my normal scan area that runs accounts at near full capacity - ie takes a long time to catch up).

It might also have to do with my doing a lucky egg evolution spree on my main account on the same IP before I started having problems. I did 126 evolutions (and therefore 126 app close and relaunch and login) in 30mins.

I'm using my own internet connection.

**Comment by rafsawicki** [comment]
Today I also ran into a mass soft bans issue for the first time ever. The map was running fine until I moved it to a new server (sharing same IP), so I had to go through a catch-up phrase. Near the end of first part I started getting a lot of empty cell messages. I'm using over 80 workers to scan 10264 spawn points ( continuous range of 50 hex layers) and previously it was more than enough to avoid any issues.

**Comment by c00ni** [comment]
Just got a notification for a 13m 30s Pokémon which means the scanner (which is now back to my normal scan area, not the massive one)  is running behind.
Checked and I'm getting a few non-empty cells returned as empty but _not all accounts are doing this now_.
I'll reintroduce the 10s delay to my encounters, but I'd literally been running the current setup for days without issue.

**Comment by phattbeats** [comment]
Coming here to report the same thing. My current setup of ~80 accounts across 4 IDs will now be softbanned after about 8 hours of continuous run. I have been running scanners of my area for about a month. Any tips on how to run from multiple IPs?

**Comment by darkforcesjedi** [comment]
It doesn't matter. The softbans have begun on my accounts too now. Within the last 15 minutes nearly all of mine have gone down.

**Comment by phattbeats** [comment]
That is... unfortunate.

**Comment by darkforcesjedi** [comment]
It looks like they have done something within the last couple hours to kill scanners.

**Comment by c00ni** [comment]
Yep - see https://www.reddit.com/r/pokemongodev/comments/55zykk/scan_limits/

**Comment by darkforcesjedi** [comment]
It looks like the SafetyNet validation has changed as well. I can no longer connect on my normal account on my phone (rooted Note 5 with suhide). No point in keeping a map up anyway if I can't play the game.

**Comment by seikur0** [comment]
#285 That's probably it. Today it happened for me as well after some hours of scanning.
